halo2_proofs/src/dev.rs

```diff
@@ -1169,13 +1171,15 @@ impl<F: FromUniformBytes<64> + Ord> MockProver<F> {
     /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
     /// the reasons that the circuit is not satisfied.
     /// Constraints and lookup are checked at `usable_rows`, parallelly.
+    #[cfg(feature = "multicore")]
```

Comment:
It's mainly because `maybe_rayon` doesn't have all traits we need when `maybe_rayon/threads` is disabled, so it's not that useful tbh.

Also I feel it makes sense for this function to only exist when `multicore` is enabled, for application use `halo2_proofs` without `multicore` could just use the original `verify` function.